code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -834,30 +832,19 @@ GLOBAL_LIST_EMPTY(dynamic_station_traits)
 	if (!isnull(threat_log))
 		log_threat(-cost, threat_log, reason)
 
-/// Turns the value generated by lorentz distribution to number between 0 and 100.
-/// Used for threat level and splitting the budgets.
-/datum/game_mode/dynamic/proc/lorentz_to_amount(x)
-	switch (x)
-		if (-INFINITY to -20)
-			return rand(0, 10)
-		if (-20 to -10)
-			return RULE_OF_THREE(-40, -20, x) + 50
-		if (-10 to -5)
-			return RULE_OF_THREE(-30, -10, x) + 50
-		if (-5 to -2.5)
-			return RULE_OF_THREE(-20, -5, x) + 50
-		if (-2.5 to -0)
-			return RULE_OF_THREE(-10, -2.5, x) + 50
-		if (0 to 2.5)
-			return RULE_OF_THREE(10, 2.5, x) + 50
-		if (2.5 to 5)
-			return RULE_OF_THREE(20, 5, x) + 50
-		if (5 to 10)
-			return RULE_OF_THREE(30, 10, x) + 50
-		if (10 to 20)
-			return RULE_OF_THREE(40, 20, x) + 50
-		if (20 to INFINITY)
-			return rand(90, 100)
+/**
+ * Returns the comulative distribution of threat centre and width, and a random location of -0.5 to 0.5
+ * plus or minus the otherwise unattainable lower and upper percentiles. All multiplied by the maximum
+ * threat and then rounded to the nearest interval.
+ */
+/datum/game_mode/dynamic/proc/lorentz_to_amount(centre, scale, max_threat = 100, interval = 1)
+	. = LORENTZ_CUMULATIVE_DISTRIBUTION(centre, rand()-0.5, scale)
+	/**
+	 * Given an x variable value of -/+5, a location of -/+ 0.5, a scale of 0.5 to 4, and a graphing calculator,
+	 * the threat will go beyond the upper and lower 3% of the max threat, which is why
+	 * we're leaving the remaining 3% to the random number generator.
+	 */
+	return round(. * max_threat + rand(-3, 3) * max_threat/100, interval)
```

Comment:
```suggestion
	/**
	 * Given an x variable value of -/+5, a location of -/+ 0.5, a scale of 0.5 to 4, and a graphing calculator,
	 * the threat will go beyond the upper and lower 3% of the max threat, which is why
	 * we're leaving the remaining 3% to the random number generator.
	 */
	return round(LORENTZ_CUMULATIVE_DISTRIBUTION(centre, rand()-0.5, scale) * max_threat + rand(-3, 3) * max_threat/100, interval) 
```

if you think this is too long or ugly, we can just do this

```suggestion
	var/lorentz_result = LORENTZ_CUMULATIVE_DISTRIBUTION(centre, rand()-0.5, scale)
	/**
	 * Given an x variable value of -/+5, a location of -/+ 0.5, a scale of 0.5 to 4, and a graphing calculator,
	 * the threat will go beyond the upper and lower 3% of the max threat, which is why
	 * we're leaving the remaining 3% to the random number generator.
	 */
	return round(lorentz_result * max_threat + rand(-3, 3) * max_threat/100, interval)
```

the dot variable is no longer useful regardless and should be changed to something else because it's superfluous and un-needed. would also prefer you add parens to this so that the meaning doesn't get lost because I think it's meant to be `lorentz_result * max_threat + rand(-3, 3) * (max_threat/100)` but someone else could easily confuse it for `(lorentz_result * max_threat + rand(-3, 3) * max_threat)/100` or think that's what was intended and... argh just tweak it